pkg/driver/node_test.go

```diff
@@ -616,8 +653,159 @@ func TestNodeUnpublishVolume(t *testing.T) {
 				mockMounter.EXPECT().Unmount(targetPath).Return(tc.unmountReturn)
 			}
 
+			if tc.setupVolUsageCache {
+				volUsageCache = make(map[string]*volMetrics)
+				volUsageCache[targetPath] = metrics
+			}
+
 			ret, err := driver.NodeUnpublishVolume(ctx, tc.req)
 			testResult(t, "NodeUnpublishVolume", ret, err, tc.expectError)
 		})
 	}
 }
+
+func TestNodeGetVolumeStats(t *testing.T) {
+	var (
+		validPath   = "/tmp/target"
+		invalidPath = "/path/does/not/exist"
+		volMetrics  = &volMetrics{
+			volPath:   validPath,
+			timeStamp: time.Now().Add(time.Duration(-10) * time.Minute),
+			volUsage: []*csi.VolumeUsage{
+				{
+					Unit:      csi.VolumeUsage_BYTES,
+					Available: 1,
+					Used:      1,
+					Total:     2,
+				},
+			},
+		}
+	)
+	makeDir(validPath)
```

Comment:
test environment doesn't like this, not sure if https://golang.org/pkg/io/ioutil/#TempDir will work either